Search topic reorg and new nav menus

alicejw-aws opened the pull request:
### Description
Create infra for new categories of search topics. Create left nav menus and reorganize files accordingly.

### Issues Resolved
[2135](https://github.com/opensearch-project/documentation-website/issues/2135)

Resolves part of META for entire doc set:
[1036](https://github.com/opensearch-project/documentation-website/issues/1036)

### Rendered site images
Updated 
1. Search and query menu (collapsed):
![Screen Shot 2022-12-07 at 11 42 29 AM](https://user-images.githubusercontent.com/88908598/206279961-fedc7e5b-05c1-4be4-8bc9-fd1e93effd10.png)
2. Search and query menu (expanded): 
![search-menu-expanded](https://user-images.githubusercontent.com/88908598/206302767-12955a96-1fab-4863-9a67-13bb2370e1bf.png)

3. New Search overview. 
**Images:**
![Search-overview2](https://user-images.githubusercontent.com/88908598/206304885-ecdd15dc-9f7c-46fa-89d1-ef994cbec6bd.png)
![Search-overview1](https://user-images.githubusercontent.com/88908598/206304875-ea0fd2cb-280a-4e1e-9ba9-a061497a0a08.png)

**PDF:**
[search-overview.pdf](https://github.com/opensearch-project/documentation-website/files/10180294/search-overview.pdf)






### Checklist
- [x ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Naarcha-AWS commented:
@alicejw-aws: Can we put this as all one collection instead of adding multiple new ones?

Naarcha-AWS commented:
Please put this all into one collection titled "Search and Query". That way, all of the types of Search can be subsections of the main collection. For example:

---
Search and query
   - Search overview
   - Custom search options
   - Search preparation 
   - Query DSL
---

alicejw-aws commented:
> Please put this all into one collection titled "Search and Query". That way, all of the types of Search can be subsections of the main collection. For example:
> 
> Search and query
> 
> * Search overview
> * Custom search options
> * Search preparation
> * Query DSL

This structure was developed about a month ago offline in a quip document. This PR reflects the infrastructure of menu collections in the config file, and new front matter for individual pages. 
We probably need a compelling reason to change it after the work is now completed. 

The developed the new categories to organize topics logically for the user to find them. 
This is for SEO and discoverability. Thanks

alicejw-aws commented:
Update: reworking structure of config YAML and all individual pages' front matter now, per feedback request.

**Search and query** - single collection/menu
Individual category menus -> convert to parent page with child pages for that section
**Search your data**
   Search overview
   ...
**Prepare data for searches**
   Index data
   ...
...and so on.

alicejw-aws commented:
Update: Finished updates on page restructuring. Ready for doc review.

Naarcha-AWS commented on _opensearch/pipeline-agg.md:
```diff
@@ -1,8 +1,8 @@
 title: Pipeline aggregations
-parent: Aggregations
-nav_order: 4
+parent: Aggregate data for analysis
```
Can we move these files to the Search and Query directory?

alicejw-aws replied:
ok sure

Naarcha-AWS commented on _search-plugins/async/index.md:
```diff
@@ -1,13 +1,13 @@
 ---
 layout: default
-title: Asynchronous search
+title: Asynchronous search plugin
```
Should we remove the `search_plugin` directory and collection? For example, this page would go into the following directory, `_search-query` > `async` > `index.md`. Then, we can place any child pages of "Asynchronous search plugin" into the `async` directory.

alicejw-aws replied:
would that make sense to move all of the search plugins back to Search and query? We'd have to move them all (k-NN, quergy, SQL, async).

just to provide context. . .In the original plan we had these categories under Search and query: 
**Search with SQL**
**Search asynchronously** 
but because they are actual plugins, i thought best to leave them within the **Search plugins** menu. 
but if that makes sense to move all search plugins to Search and query we can do that.

hdhalter replied:
They need to be moved under the Search category. We will not have any "x plugin" categories in the nav pane going forward.

alicejw-aws replied:
ok sounds good.

Naarcha-AWS commented on _search-plugins/knn/index.md:
```diff
@@ -1,14 +1,14 @@
 ---
 layout: default
-title: k-NN
+title: k-NN plugin
```
See my previous comment about the Asynchronous plugin page.

Naarcha-AWS commented on _search-plugins/querqy/index.md:
```diff
@@ -1,13 +1,13 @@
 ---
 
-# Querqy
+# Querqy plugin
```
Same thing here as kNN and Asynchronous search.

Naarcha-AWS commented on _search-query/text-analyzers.md:
```diff
@@ -1,12 +1,12 @@
-title: Text analyzers
-parent: Query DSL
-nav_order: 41
+title: Optimize text for searches
```
Make sure that we make a directory for this page and all of this page's children.

alicejw-aws replied:
ok i can do that again. that's how i had it prior to the single menu feedback. 

> We need to clean up the directory structure of this PR before we merge it in. The changes include:
> 
> * Move any pages under the `_opensearch` directory that have been changed to the `_search-and-query` directory.
This was done after the first round of feedback you provided. 
Will do for the new ones ( Aggregate data for analysis category).
> * Create sub-directories under `_search-and-query` for each Parent section, including: "Asynchronous search plugin", "Search using Query DSL", "Search using kNN", and "Prepare data for searches".
Got it. Will build those out and move to Search
> * Move any edited pages from `_search-plugins` into the `_search-and-query` directory.
> * Remove `search-plugins` from collections.
Yes.
Sounds good. I can build the alternate structures you suggested.

Naarcha-AWS reviewed (changes requested):
We need to clean up the directory structure of this PR before we merge it in. The changes include:

- Move any pages under the `_opensearch` directory that have been changed to the `_search-and-query` directory.
- Create sub-directories under `_search-and-query` for each Parent section, including: "Asynchronous search plugin", "Search using Query DSL", "Search using kNN", and "Prepare data for searches".
- Move any edited pages from `_search-plugins` into the `_search-and-query` directory.
- Remove `search-plugins` from collections.

alicejw-aws commented:
Update: sending for PM (Stavros) to review the Search overview content. 
The goal is to have an overview that covers all available search features, plugins, and options so that any current or future OpenSearch user can get a sense of what we provide when they read this page.

alicejw-aws commented:
To address some feedback questions received outside of the PR: 
Here are some reasons, and we can reassess once everyone can provide an opinion on them before we move the files.

1. Query DSL - reasons why it should be its own left navigation menu parallel to OpenSearch. Agree with these reasons: 
There are multiple future topics to add to this section, so it makes sense to have it visible in left navigation. Will also gather PM's input on TOC structure for this category of topics.
2. Question: should **Optimize text for searches** and **Aggregate data for analysis** categories and topics be nested under Search and query? 
Yes, they are search preparation tasks. 
3. Could we get cross-fn team to review the new Search overview? Yes. It is in review with PM now.
4. Custom search features such as autocomplete, and paginate - was suggested to have a separate overview page with links to topic pages. 
(No, IMO): this does belong in the comprehensive Search overview with links to the individual topic pages. We want customers to get the full story of what OpenSearch provides for search functionality on one page.

alicejw-aws commented:
hi @Naarcha-AWS , I'm making the current changes you requested now. 

> We need to clean up the directory structure of this PR before we merge it in. The changes include:
> 
> * Move any pages under the `_opensearch` directory that have been changed to the `_search-and-query` directory.
This was done after the first round of feedback you provided. 
Will do for the new ones ( Aggregate data for analysis category).
> * Create sub-directories under `_search-and-query` for each Parent section, including: "Asynchronous search plugin", "Search using Query DSL", "Search using kNN", and "Prepare data for searches".
Got it. Will build those out and move to Search
> * Move any edited pages from `_search-plugins` into the `_search-and-query` directory.
> * Remove `search-plugins` from collections.
Yes.
Sounds good. I can build the alternate structures you suggested.

alicejw-aws commented:
Creating subdirectories for each category under Search and query now. That was a good catch, thanks!

alicejw-aws commented:
Updates are now complete for @Naarcha-AWS 's review. 

One note: we might want to also rename the "x plugin" category titles to be parallel with other Search and query categories: 

Async search plugin -> Search asynchronously
k-NN plugin -> Search for k-nearest neighbors
Quergy plugin -> Search with quergy

* will make changes after doc reviewers provide suggestion/opinion on that. 
Thanks

alicejw-aws commented:
Updates: 
1. PM confirmed he is reviewing the overview now. 
2. I fixed the merge conflict (add new file that was checked into main) for geo-bounding-box.md in git CLI: 
git checkout main _opensearch/query-dsl/geo-bounding-box.md
and updated the new parent title for the file.

alicejw-aws commented:
Update: I found 94 broken cross-references, so I fixed about half of them. There are now 53 links TBD.
for the broken links in external files to this PR, I left those to fix separately (just to avoid creating merge conflicts).

alicejw-aws commented:
Update: all cross-references between files in this PR have been updated. 

Note: there are 37 remaining cross-references that need to be updated in other files external to this PR. I thought it would be best to created a separate PR to fix those. Thanks

alicejw-aws commented:
Update: all cross-references to other pages that link to pages that changed with this new search query directory structure are now updated.
Total: 94 fixed

hdhalter commented:
Nate is making final changes. I'll reassign to him.

alicejw-aws commented:
Update: Jan. 5 - @Naarcha-AWS is recreating a separate PR for this.

Naarcha-AWS commented:
Closing this to open a new search toc PR in order to prevent merge conflicts.
